apps/src/templates/levelSummary/FreeResponseAIEvaluation.tsx

```diff
@@ -0,0 +1,128 @@
+import Button from '@code-dot-org/component-library/button';
+import React, {useEffect, useState} from 'react';
+
+import {getChatCompletionMessage} from '@cdo/apps/aiTutor/chatApi';
+import CollapsibleSection from '@cdo/apps/templates/CollapsibleSection';
+
+import SafeMarkdown from '../SafeMarkdown';
+
+import style from '@cdo/apps/levelbuilder/ai-iteration-tools/ai-tutor/ai-tutor-tester.module.scss';
+
+interface StudentResponse {
+  user_id: number;
+  text: string;
+  student_display_name: string;
+  aiEvaluation: string;
+}
+
+interface FreeResponseAIEvaluationProps {
+  responses: StudentResponse[];
+  levelInstructions: string;
+}
+const FreeResponseAIEvaluation: React.FunctionComponent<
+  FreeResponseAIEvaluationProps
+> = ({responses, levelInstructions}) => {
+  const [evaluationsPending, setEvaluationsPending] = useState<boolean>(false);
+  const [evaluations, setEvaluations] = useState<StudentResponse[]>([]);
+  const [evaluationCount, setEvaluationCount] = useState<number>(0);
+  const [aiSummary, setAiSummary] = useState<string>('');
+  const evaluationComplete = responses.length === evaluationCount;
+
+  useEffect(() => {
+    if (evaluationComplete) {
+      setEvaluationsPending(false);
+      summarizeStudentEvaluations(evaluations);
+    }
+  }, [evaluations, evaluationComplete]);
+
+  const basePrompt =
+    'You are a teaching assistant for a high school AP Computer Science class where the students are learniing JavaScript.';
+
+  const getAIEvaluations = async () => {
+    setEvaluationsPending(true);
+    const responsePromises = responses.map(async studentResponse => {
+      return evaluateStudentResponse(studentResponse);
+    });
+
+    await Promise.allSettled(responsePromises);
+  };
+
+  const evaluateStudentResponse = async (studentResponse: StudentResponse) => {
+    const studentPrompt = `${basePrompt} Please review the student's responses and indicate whether the response is "great", "ok", or "needs revision". Provide one sentence with your reasoning. The student's instructions are: ${levelInstructions}.`;
```

Comment:
No action needed for this PR - more just a wondering thought:

I am sure this is in follow up work, but right now this wouldn't work with free response questions where there is an image included in the instructions, correct?  I am guessing there will be an added condition for if AI analysis is available for the question asked.

Reply:
oooh interesting! Yeah, this isn't set up at all to handle a case where there is an image in the instructions. And I think you're right, we just wouldn't (at least initially) make AI Evaluation available in that case. Do you happen to have a specific level in mind? 

If and when we cross that bridge, my guess is that there are two basic paths: 1.) Leverage rubrics (or something rubric-y) to give the AI examples of different buckets ("great", 'good", "ok" or whatever) of answers so it doesn't need to know specifically about the image. We might have to do this to hit accuracy goals for levels without images in the instructions any way. Or if that doesn't work 2.) see what the Student Learning team figures out about image processing via AI and make a plan from there.

Reply:
Here's an example:

https://studio.code.org/s/csp4-2024/lessons/7/levels/9

I think the curriculum team has been trying to move away from images like this because of accessibility and translatability issues, but my guess is that there's still a handful out there.

Reply:
Found a free response one with an image:  https://studio.code.org/s/csp4-2024/lessons/2/levels/7